File: datamodel/src/main/java/datamodel/clinical/VitalFunction.java
package datamodel.clinical;

import java.time.LocalDate;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class VitalFunction {

    @NotNull
    public abstract LocalDate date();

    @NotNull
    public abstract VitalFunctionCategory category();

    @NotNull
    public abstract String subcategory();

    public abstract double value();

    @NotNull
    public abstract String unit();
}

File: algo/src/main/kotlin/com/hartwig/actin/datamodel/clinical/VitalFunction.kt
package com.hartwig.actin.datamodel.clinical

import java.time.LocalDateTime

data class VitalFunction(
    val date: LocalDateTime,
    val category: VitalFunctionCategory,
    val subcategory: String,
    val value: Double,
    val unit: String,
    val valid: Boolean
)
